src/coreComponents/physicsSolvers/wavePropagation/AcousticTTIFletcherWaveEquationSEMKernel.hpp

```diff
@@ -0,0 +1,641 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOS Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file AcousticTTIFletcherWaveEquationSEMKernel.hpp
+ */
+
+#ifndef GEOS_PHYSICSSOLVERS_WAVEPROPAGATION_ACOUSTICTTIFLETCHERWAVEEQUATIONSEMKERNEL_HPP_
+#define GEOS_PHYSICSSOLVERS_WAVEPROPAGATION_ACOUSTICTTIFLETCHERWAVEEQUATIONSEMKERNEL_HPP_
+
+#include "finiteElement/kernelInterface/KernelBase.hpp"
+#include "WaveSolverUtils.hpp"
+#if !defined( GEOS_USE_HIP )
+#include "finiteElement/elementFormulations/Qk_Hexahedron_Lagrange_GaussLobatto.hpp"
+#endif
+#include "AcousticTTIFields.hpp"
+
+
+namespace geos
+{
+
+/// Namespace to contain the acoustic TTI (Fletcher's equations) wave kernels .
+namespace acousticTTIFletcherWaveEquationSEMKernels
+{
+
```

Comment:
Can you take a look to PRs, #2995 and #3032 to adapt your implementation with the rafactoring work?